data-model/src/main/java/io/micronaut/data/model/schema/Column.java

```diff
@@ -0,0 +1,426 @@
+/*
+ * Copyright 2017-2025 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.data.model.schema;
+
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.data.annotation.GeneratedValue;
+import io.micronaut.data.exceptions.MappingException;
+import io.micronaut.data.model.DataType;
+import io.micronaut.data.model.JsonDataType;
+import io.micronaut.data.model.query.builder.sql.Dialect;
+
+
+/**
+ * Represents a database column from mapped entity persistent field.
+ *
+ * @author radovanradic
+ * @since 4.13.0
+ */
+@Internal
+public final class Column {
```

Comment:
can we name this something else? Like `ColumnDefinition` or something since we already have annotated called `@Column` and users may import the wrong then which is annoying